samplers.go

```diff
@@ -112,6 +123,32 @@ func (s *Set) Flush() []DDMetric {
 	}}
 }
 
+func (s *Set) Export() (JSONMetric, error) {
+	val, err := s.hll.GobEncode()
+	if err != nil {
+		return JSONMetric{}, err
+	}
+	return JSONMetric{
+		MetricKey: MetricKey{
+			Name:       s.name,
+			Type:       "set",
+			JoinedTags: strings.Join(s.tags, ","),
+		},
+		Tags:  s.tags,
+		Value: val,
+	}, nil
+}
+
+func (s *Set) Combine(other []byte) error {
+	// you can only merge hyperloglogs with the same compression parameter
```

Comment:
I was amusingly reading this (up to this point) and thought "ugh what if someone runs instances with differing args…" 😃 👍